dora/core/common/src/main/java/alluxio/uri/UfsUrl.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.uri;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.InvalidPathException;
+import alluxio.grpc.UfsUrlMessage;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class UfsUrl {
+
+  public static final String SCHEME_SEPERATOR = "://";
+  public static final String PATH_SEPERATOR = "/";
+  UfsUrlMessage mProto;
```

Comment:
The java `UfsUrl` object should not rely on a Grpc representation of a URL, so that we can get rid of a dependency on protobuf and gRPC. Store scheme, authority and path components as native Java types.